src/util/symbol_table.h

```diff
@@ -122,15 +122,8 @@ class symbol_tablet : public symbol_table_baset
 
   typedef symbolst::const_iterator const_iteratort;
 
-  virtual const_iteratort begin() const
-  {
-    return internal_symbols.begin();
-  }
-
-  virtual const_iteratort end() const
-  {
-    return internal_symbols.end();
-  }
+  using symbol_table_baset::begin;
+  using symbol_table_baset::end;
```

Comment:
develop `symbol_table_baset::begin/end` are both purely virtual, so can not be used in this way.

Reply:
Discussed out of band: more consistent to have both const and non-const iterators be virtual and defined in the base class.